ports/rp2/machine_spi.c

```diff
@@ -167,7 +180,12 @@ mp_obj_t machine_spi_make_new(const mp_obj_type_t *type, size_t n_args, size_t n
         }
         self->mosi = mosi;
     }
-    if (args[ARG_miso].u_obj != mp_const_none) {
+
+    if (args[ARG_miso].u_obj == MP_ROM_INT(-1)) {
+        self->miso = spi_id == 0 ? MICROPY_HW_SPI0_MISO : MICROPY_HW_SPI1_MISO;
```

Comment:
In this case of -1, `miso` is not given and so the pin should not change.

The default is already set in the static SPI instance, so this conditional is not needed.